ci(micropython): fix MicroPython github build action

PGNetHun opened the pull request:
Fixes #7940 

Update LVGL MicroPython github build action.

[lv_micropython](https://github.com/lvgl/lv_micropython) and [lv_binding_micropython](https://github.com/lvgl/lv_binding_micropython) repos are updated to use [MicroPython version 1.24.1](https://github.com/micropython/micropython/tree/v1.24.1)

LVGL is used as [MicroPython external/user C module](https://docs.micropython.org/en/latest/develop/cmodules.html), so CI build is also updated.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

kisvegabor reviewed (dismissed):
Thank you! It seems it should be preferred over #7943

kisvegabor commented:
By coincidence the Windows tests broke at the same time, but it's unrelated.

PGNetHun commented:
STM32 port still [fails](https://github.com/lvgl/lvgl/actions/runs/13910693435/job/38929673725?pr=7945), I will fix it soon.

lhdjply commented:
![image](https://github.com/user-attachments/assets/b7261c22-7f09-41f9-a877-ef901c4cb160)
Can `esp32_idf44_setup` be changed to `esp32_idf_setup` and run on Ubuntu 24.04?   cc @liamHowatt

PGNetHun commented:
> ![image](https://private-user-images.githubusercontent.com/18370433/423737005-b7261c22-7f09-41f9-a877-ef901c4cb160.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIyODA5MTIsIm5iZiI6MTc0MjI4MDYxMiwicGF0aCI6Ii8xODM3MDQzMy80MjM3MzcwMDUtYjcyNjFjMjItN2YwOS00MWY5LWE4NzctZWY5MDFjNGNiMTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE4VDA2NTAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhYWZiMmViYzY1NjNmYTJkYjc0ODA2MmQ2ODI2NGJiZjUwYjJjMmQ0ODU3NGU5ZWUwYzRlZDY4YTQ1ODMxZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.19mIihpEMzBFFb8X1kr0tzNWncEfY4NRiMCQMv3wtG0) Can `esp32_idf44_setup` be changed to `esp32_idf_setup` and run on Ubuntu 24.04? cc @liamHowatt

MicroPython [requires ESP-IDF 5.x version](https://github.com/micropython/micropython/blob/master/ports/esp32/README.md#setting-up-esp-idf-and-the-build-environment), and it's [ci.sh file](https://github.com/micropython/micropython/blob/master/tools/ci.sh#L149) is also changed, so that's why I've changed ESP32 build in our workflow file too.

lhdjply commented:
![image](https://github.com/user-attachments/assets/1ec49134-d9b2-4d75-b803-c6b674449235)

My question is, after changing **esp32_idf44_setup** to **esp32_idf_setup**, will it successfully run on **Ubuntu 24.04**? If so, can you upgrade the **Ubuntu 22.04** environment originally set up for **ESP32** to **Ubuntu 24.04**? Previously, since **esp32_idf44_setup** failed to compile on **Ubuntu 24.04**, I set up a separate **Ubuntu 22.04** environment for **ESP32**.

PGNetHun commented:
> My question is, after changing **esp32_idf44_setup** to **esp32_idf_setup**, will it successfully run on **Ubuntu 24.04**? If so, can you upgrade the **[Ubuntu 22.04](https://github.com/lvgl/lvgl/blob/6f8a30eb3eb748a814fa80e079586d35e9d26159/.github/workflows/build_micropython.yml#L87)** environment originally set up for **ESP32** to **Ubuntu 24.04**? Previously, since **esp32_idf44_setup** failed to compile on **Ubuntu 24.04**, I set up a separate **Ubuntu 22.04** environment for **ESP32**.

Oh, sorry, yes the build works on Ubuntu 24, I'm using that also, and changed in [lv_binding_micropython](https://github.com/lvgl/lv_binding_micropython/blob/master/.github/workflows/unix_port.yml#L10) repo too.
Thanks for pointing that, I have to change in [lv_micropython repo](https://github.com/lvgl/lv_micropython/blob/master/.github/workflows/ports_esp32.yml#L28) also.

liamHowatt reviewed (dismissed):
Ah, great, thank you. I didn't know it would be this simple. But it would've taken me time to realize `PICO` -> `RPI_PICO`

I opened a PR to fix the unrelated Windows CI issue: #7948

PGNetHun commented:
> Ah, great, thank you. I didn't know it would be this simple. But it would've taken me time to realize `PICO` -> `RPI_PICO`
> 
> I opened a PR to fix the unrelated Windows CI issue: #7948

1. CI build script suggested to change that obsolete name to `RPI_PICO` :-)
2. Thanks for fixing Windows build issue

CI builds pass, ready to merge.


PS:
next step for me is to clean up `lv_micropython` repo `.github` folder: remove unnecessary files

kisvegabor commented:
Thank you!
